### PR TITLE
Add assert_error_covers(expected, fn, ...)

### PR DESCRIPTION
**Pull request by nshy**
It checks that `fn` raises error of type table that includes `expected` table. Altenatively error can be Tarantool box error. In this case error in converted to table using `error:unpack()` first.

Required for https://github.com/tarantool/tarantool/issues/9111

### CODE REVIEW

File: luatest/assertions.lua [hunk 1]
```diff
@@ -563,6 +563,23 @@ function M.assert_error_msg_matches(pattern, fn, ...)
     end
 end
 
+function M.assert_error_covers(expected, fn, ...)
```
Comment (ylobankov) [inline luatest/assertions.lua:566]: Please add a doc string above.

Comment (nshy) [reply luatest/assertions.lua:566]: Fixed.

```diff
diff --git a/luatest/assertions.lua b/luatest/assertions.lua
index db6626c..9ee1791 100644
--- a/luatest/assertions.lua
+++ b/luatest/assertions.lua
@@ -563,6 +563,14 @@ function M.assert_error_msg_matches(pattern, fn, ...)
     end
 end

+--- Checks that error raised by function is table that includes expected one.
+---
+--- If error object supports unpack() method (like Tarantool errors) then
+--- error is unpacked to get the table to be compared.
+--
+-- @tab expected
+-- @func fn
+-- @param ... arguments for function
 function M.assert_error_covers(expected, fn, ...)
     local ok, actual = pcall(fn, ...)
     if ok then
```

**Review by ylobankov — APPROVED** [review]

File: test/luaunit/assertions_error_test.lua [hunk 2]
```diff
@@ -118,3 +118,26 @@ function g.test_assert_errorMsgMatches()
...
+        e.unpack = function()
+            return {a = a, b = b}
+        end
+        error(e)
+    end
+    -- good error coverage, error has unpack()
+    t.assert_error_covers({b = 2}, error_with_unpack, 1, 2)
+    -- bad error coverage
+    assert_failure(t.assert_error_covers, {b = 2},
+                     function(a, b) error({a = a, b = b}) end, 1, 3)
```
Comment (ylobankov) [inline test/luaunit/assertions_error_test.lua:142]: Please fix indention.

Comment (nshy) [reply test/luaunit/assertions_error_test.lua:142]: Fixed.

```diff
diff --git a/test/luaunit/assertions_error_test.lua b/test/luaunit/assertions_error_t>
index 67db1ae..a385f30 100644
--- a/test/luaunit/assertions_error_test.lua
+++ b/test/luaunit/assertions_error_test.lua
@@ -139,5 +139,5 @@ function g.test_assert_errorCovers()
     t.assert_error_covers({b = 2}, error_with_unpack, 1, 2)
     -- bad error coverage
     assert_failure(t.assert_error_covers, {b = 2},
-                     function(a, b) error({a = a, b = b}) end, 1, 3)
+                   function(a, b) error({a = a, b = b}) end, 1, 3)
 end
 end
```

**Pull request merged**